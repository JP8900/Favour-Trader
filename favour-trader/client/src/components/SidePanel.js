import React, { Component } from 'react';
import Dock from 'react-dock';
import SidePanelContent from './SidePanelContent';

class SidePanel extends Component {
  render() {
    const { isVisible, authService } = this.props;
    const dockStyle = {
      width: '0px',
      marginTop: '55px',
      backgroundColor: '#9978C2',
      overflow: 'hidden',
    };

    return (
      <Dock size={0.2} position='left' isVisible={isVisible} dimMode={'none'} dockStyle={dockStyle}>
      {
        (authService.loggedIn()) ? (
          <SidePanelContent authService={authService}/>
        ) : (
          ''
        )
      }
      </Dock>
    );
  }
}

export default SidePanel;