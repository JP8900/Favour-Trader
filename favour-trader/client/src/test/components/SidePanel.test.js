import React from 'react';
import renderer from 'react-test-renderer';
import SidePanel from '../../components/SidePanel.js';
import AuthServiceMock from './AuthServiceMock.js';

it('renders correctly', () => {
  var authServiceMock = new AuthServiceMock();
  const tree = renderer
    .create(<SidePanel isVisible={true} authService={authServiceMock} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});