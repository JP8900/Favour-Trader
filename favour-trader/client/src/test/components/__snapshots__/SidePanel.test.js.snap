// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "height": 0,
      "left": 0,
      "position": "fixed",
      "top": 0,
      "width": 0,
      "zIndex": 99999999,
    }
  }
>
  <div
    style={
      Object {
        "MozBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "MozTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out",
        "OBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "OTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out",
        "WebkitBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "WebkitTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out",
        "background": "white",
        "backgroundColor": "#9978C2",
        "boxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "height": "100%",
        "left": 0,
        "marginTop": "55px",
        "msBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "msTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out",
        "overflow": "hidden",
        "position": "fixed",
        "top": 0,
        "transition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out",
        "width": "20%",
        "zIndex": 1,
      }
    }
  >
    <div
      onMouseDown={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "opacity": 0,
          "position": "absolute",
          "right": -5,
          "top": 0,
          "width": 10,
          "zIndex": 2,
        }
      }
    />
    <div
      style={
        Object {
          "height": "100%",
          "overflow": "auto",
          "width": "100%",
        }
      }
    >
      
    </div>
  </div>
</div>
`;
